### PR TITLE
do release in github actions

### PR DESCRIPTION
**Pull request by ry**
using https://github.com/marketplace/actions/gh-release

towards https://github.com/denoland/deno/issues/2979

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -108,5 +108,29 @@ jobs:
...
+        uses: softprops/action-gh-release@v1
+        if: matrix.kind == 'test' && startsWith(github.ref, 'refs/tags/')
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          files: |
+            target/release/deno_win_x64.zip
+            target/release/deno_linux_x64.gz
+            target/release/deno_osx_x64.gz
+          draft: true
```
Comment (piscisaureus) [inline .github/workflows/build.yml]: this stays?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -108,5 +108,29 @@ jobs:
...
+
+      - name: Pre-release (windows)
+        if: startsWith(matrix.os, 'windows')
+        run: PowerShell -Command "& {Compress-Archive -CompressionLevel Optimal -Force -Path target/release/deno.exe -DestinationPath target/release/deno_win_x64.zip}"
+
+      - name: Release
+        uses: softprops/action-gh-release@v1
+        if: matrix.kind == 'test' && startsWith(github.ref, 'refs/tags/')
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (piscisaureus) [inline .github/workflows/build.yml]: Surprised this works, I thought it had to be in the "with" section to be passed as an action param.

File: .github/workflows/build.yml (hunk 1 above)
Comment (ry) [reply .github/workflows/build.yml]: Yes - two reasons

1. I always manually edit the release text to include download links and the changes, so having it in draft mode helps.

2. If one binary (say osx) is deployed before the others, then the install.sh script will detect it as the latest release on all platforms - which results in 404s until the other binaries have been posted. Draft solves this.

File: .github/workflows/build.yml (hunk 2 above)
Comment (ry) [reply .github/workflows/build.yml]: *shrug*

**Pull request merged**